core/sql/executor/ex_hash_grby.cpp

```diff
@@ -452,6 +463,8 @@ short ex_hash_grby_tcb::work() {
             hasFreeTupp_ = TRUE;
     	};
       };
+      if (bmoStats_)
+         bmoStats_->setBmoPhase(PHASE_END-HGB_READ_PHASE);
```

Comment:
With the lack of test case to test out the phase changes, I took a safer approach of setting it many times. I will move this code to the two places mentioned after I did some testing with TPCDS queries